EQTool/Services/Parsing/LogStartCustomTimer.cs

```diff
@@ -27,61 +56,70 @@ public bool Handle(string line, DateTime timestamp)
             return false;
         }
 
-        public CustomTimer GetStartTimer(string message)
+        //
+        // parse the passed line to see if user has requested a custom timer to started
+        //
+        public CustomTimer ParseStartTimer(string line)
         {
-            var timer = GetStartTimer(message, StartTimer);
-            if (timer != null)
-            {
-                return timer;
-            }
-            else
+            // return value
+            CustomTimer rv = null;
+
+            // set up a crazy regex to watch for formats like
+            //      PigTimer-30                  30 second timer, with no name
+            //      PigTimer-120-description     120 second timer, with name 'description'
+            //      PigTimer-6:40-Guard          6 minutes 40 second timer, with name 'Guard'
+            //      PigTimer-10:00               10 minute timer, with no name
+            //      PigTimer-1:02:00-LongTimer   1 hour, 2 minute timer, with description 'LongTimer'
+            string customTimerPattern = @"PigTimer-(((?<hh>[0-9]+):)?((?<mm>[0-9]+):))?(?<ss>[0-9]+)(-(?<label>\w+))*";
+            //                            PigTimer-                                                                         must start with PigTimer-
+            //                                      ((?<hh>[0-9]+):)?                                                       Group "hh"      0 or more (sets of numbers, followed by a colon)
+            //                                                       ((?<mm>[0-9]+):)                                       Group "mm"      1 set of numbers followed by a colon
+            //                                     (                                 )?                                                     0 or more hh and mm groups
+            //                                                                         (?<ss>[0-9]+)                        Group "ss"      1 set of numbers
+            //                                                                                      (-(?<label>\w+))*       Group "label"   0 or more (dash, followed by a set of word characters)
+            //
+            // This website is absolutely needed when creating this kind of crazy regex!
+            //      https://regex101.com/r/RWA9Oy/1
+            Regex regex = new Regex(customTimerPattern, RegexOptions.Compiled);
+            var match = regex.Match(line);
+            if (match.Success)
             {
-                timer = GetStartTimer(message, StartTimer1);
-            }
+                // just a little audible marker
+                System.Media.SoundPlayer player = new System.Media.SoundPlayer(@"c:\Windows\Media\chimes.wav");
```

Comment:
Please remove the playing of sound in the parsers. It makes it so when running tests locally, it actually plays sounds.
Also, it means that everyone is going to hear these sounds every time its parsed.
If we want a sounds to play when this is parse happens, a class that listens for these events can play that logic

Reply:
can do.  that's really just a debugging help for me, so I know the parser is doing it's thing